kymatio/scattering1d/backend/torch_backend.py

```diff
@@ -174,13 +174,35 @@ def swap_time_frequency(cls, x):
         x : tensor
             if complex: (batch, frequency, time, real/imag)
             else: (batch, frequency, time, 1)
+
         Returns
         -------
         output : tensor
-            if complex: (batch, time, frequency real/imag)
+            if complex: (batch, time, frequency, real/imag)
             else: (batch, time, frequency, 1)
         """
         return torch.transpose(x, dim0=-2, dim1=-3)
 
+    @staticmethod
+    def unpad_frequency(x, n1_max, n1_stride):
+        """Unpad the frequency axis after frequency scattering and/or frequency
+        averaging. This is called at the end of `jtfs_average_and_format`
+        unless `out_type='array'` and `format='joint'`.
+        NB. Unpadding is one-sided. See point 3 of pad_frequency docstring.
+        Parameters
+        ----------
+        x : tensor (batch, frequency, time, 1), corresponds to path['coef']
+        n1_max: integer. last first-order wavelet index n1 such that j1 < j2
+        for the scattering path of x. By definition, lower than len(psi1_f).
+        n1_stride: integer frequential subsampling factor associated to the
```

Comment:
`n1_stride: integer, related to the frequential subsampling factor associated to the`

`n1_stride` itself is not the frequential subsampling factor

Reply:
> n1_stride itself is not the frequential subsampling factor

i think it is? the `n1` values of `U1` are spaced by `1` whereas the `Y_fr` are spaced by `n1_stride`, so the subsampling factor is `n1_stride`

Reply:
Ah, yes, my bad.